snapcraft/models/project.py

```diff
@@ -606,42 +757,137 @@ class Project(models.Project):
-    environment: dict[str, str | None] | None = None
-    build_packages: Grammar[list[str]] | None = None
-    build_snaps: Grammar[list[str]] | None = None
-    ua_services: set[str] | None = None
-    provenance: str | None = None
-    components: dict[ProjectName, Component] | None = None
+    platforms: dict[str, Platform] | None = pydantic.Field(  # type: ignore[assignment,reportIncompatibleVariableOverride]
+        default=None,
+        description="Determines which instruction set architectures the snap builds on and runs on.",
+        examples=[
+            "platforms: {amd64: {build-on: [amd64], build-for: [amd64]}, arm64: {build-on: [amd64, arm64], build-for: [arm64]}}"
+        ],
+    )
+    assumes: UniqueList[str] = pydantic.Field(
+        default_factory=list,
+        description="The snapd features or minimum version of snapd required by the snap.",
+        examples=["assumes: [snapd2.66]", "assumes: [common-data-dir]"],
+    )
+    hooks: dict[str, Hook] | None = pydantic.Field(
+        default=None, description="Configures the snap's hooks."
+    )
+    passthrough: dict[str, Any] | None = pydantic.Field(
+        default=None,
+        description="Attributes to not validate for correctness. Useful for testing experimental snapd features.",
+    )
+    apps: dict[str, App] | None = pydantic.Field(
+        default=None,
+        description="Declares the individual programs and services that the snap runs.",
+        examples=["apps: {foo-app: {command: bin/foo-app}}"],
+    )
+    plugs: dict[str, ContentPlug | Any] | None = pydantic.Field(
+        default=None, description="Declares the snap's plugs."
+    )
+    slots: dict[str, Any] | None = pydantic.Field(
+        default=None, description="Declares the snap's slots."
+    )
+    lint: Lint | None = pydantic.Field(
+        default=None,
+        description="The linter configuration.",
+        examples=["lint: {ignore: [classic, library: [usr/lib/**/libfoo.so*]]}"],
+    )
+    epoch: str | None = pydantic.Field(
+        default=None, description="The epoch associated with this version of the snap."
+    )
+    adopt_info: str | None = pydantic.Field(
+        default=None,
+        description=textwrap.dedent(
+            """\
+        Selects a part to inherit metadata from and reuse for the snap's metadata.
+
+        Required if one of ``version``, ``summary``, or ``description`` isn't set."""
+        ),
+        examples=["adopt-info: foo-part"],
+    )
+    system_usernames: dict[str, Any] | None = pydantic.Field(
+        default=None,
+        description="The system usernames that the snap can use to run services.",
+    )
+    environment: dict[str, str | None] | None = pydantic.Field(
+        default=None, description="The snap's run-time environment variables."
+    )
+    build_packages: Grammar[list[str]] | None = pydantic.Field(
+        default=None,
+        description="Base dependent build dependencies required to build parts.",
```

Comment:
Here's my suggestion:

```suggestion
        description="The system packages required on the host so that it can build the snap.",
```

@mr-cal I don't know if there's an established preference, but I've seen "host" used throughout snapd's docs, I believe because it's more container-friendly than "system".

Reply:
I am not fond of saying build the snap, as it causes cognitive load with the parts lifecycle